libraries/resource_systemd_automount.rb

```diff
@@ -0,0 +1,21 @@
+require_relative 'resource_systemd_unit'
+require_relative 'systemd_automount'
+
+class Chef::Resource
+  class SystemdAutomount < Chef::Resource::SystemdUnit
+    self.resource_name = :systemd_automount
+    provides :systemd_automount
+
+    def unit_type(_ = nil)
+      :automount
+    end
+
+    Systemd::Automount::OPTIONS.each do |option|
+      attribute option.underscore.to_sym, kind_of: String, default: nil
+    end
+
+    def automount
```

Comment:
it's strictly an interface convenience for organization;

so this:

```ruby
systemd_service 'my-service' do
  description 'My Awesome Service'
  documentation 'man:awesome(7)'
  wanted_by 'multi-user.target'
  user 'nobody'
  group 'nobody'
  type 'oneshot'
  exec_start '/usr/bin/true'
end
```

is functionally exactly the same as:

```ruby
systemd_service 'my-service' do
  unit do
    description 'My Awesome Service'
    documentation 'man:awesome(7)'
  install do
    wanted_by 'multi-user.target'
  end
  service do
    user 'nobody'
    group 'nobody'
    type 'oneshot'
    exec_start '/usr/bin/true'
  end
end
```

so `unit`, `install`, and $unit_type attributes are totally optional, and the user can decide whether grouping attributes, and which groups to use, based on what  has the most readable presentation.